pkg/orchestration/wiring/wiringplugin/types.go

```diff
@@ -85,6 +86,16 @@ type ResourceContract struct {
 	Data   []DataItem            `json:"data,omitempty"`
 }
 
+func (c *ResourceContract) GetShape(shapeName ShapeName) (Shape, error) {
+	for _, shape := range c.Shapes {
+		if shape.Name() == shapeName {
+			return shape, nil
+		}
+	}
+
+	return nil, errors.Errorf("Shape %s not found in the contract", shapeName)
```

Comment:
It's not an error for a shape to not be there, should be just `bool`.

Reply:
Error is defensive version of panic (I don't want to use panic because it will affect all states not only the problematic ones). I assume that if this method is called the shape is 100% there and there is exactly one, that is why I didn't name the method "find". I'll add comment.